src/ownership.h

```diff
@@ -0,0 +1,18 @@
+#ifndef VCTRS_OWNERSHIP_H
+#define VCTRS_OWNERSHIP_H
+
+#include "vctrs-core.h"
+#include "utils.h"
+
+// Wrapper around `r_clone_referenced()` that only attempts to clone if
+// we indicate that we don't own `x`
+static inline SEXP vec_clone_referenced(SEXP x, const enum vctrs_ownership ownership) {
+  switch (ownership) {
+  case VCTRS_OWNERSHIP_foreign: return r_clone_referenced(x);
```

Comment:
Would be nice to have a comment along the lines of: if foreign, we only clone if not referenced. The goal is to avoid cloning temporary values that are passed to vctrs functions.

And IIRC, temporary values only have 0 refcounts if we evaluate them from C via `.External2()`?